iron-location.html

```diff
@@ -360,18 +367,29 @@
         var origin;
 
         // IE Polyfill
-        if (window.location.origin) {
-          origin = window.location.origin;
+        if (this.__location.origin) {
+          origin = this.__location.origin;
         } else {
-          origin = window.location.protocol + '//' + window.location.host;
+          origin = this.__location.protocol + '//' + this.__location.host;
         }
 
         var urlOrigin;
 
         if (url.origin) {
           urlOrigin = url.origin;
         } else {
-          urlOrigin = url.protocol + '//' + url.host;
+          // IE always adds port number on HTTP and HTTPS on <a>.host but not on
+          // window.location.host
+          var portRegex = /:[0-9]+$/;
+          var urlHost = url.host;
+          var urlHostPort = urlHost.match(portRegex);
+          var locationHostPort = this.__location.host.match(portRegex);
+
+          if (urlHostPort && !locationHostPort &&
+              (urlHostPort[0] === HTTP_PORT || urlHostPort[0] === HTTPS_PORT)) {
+                urlHost = urlHost.replace(portRegex, '');
+          }
+          urlOrigin = url.protocol + '//' + urlHost;
```

Comment:
Could this be simplified by replacing `urlHost` with `url.host.replace(/:(80|443)/, '')` (since from what I understand `window.location.host` will never have `:80` or `:443`)?

Reply:
_theoretically_ if you have `http://hostname:443` or `https://hostname:80` they should still be there. It's only when the protocol matches the port it won't be in the `window.location.host` is when it disappears.

Reply:
Ah, then maybe it can be simplified to (to avoid running multiple regexes on urlHost):

```js
var urlHost = url.host;
if (!this.__location.host.match(/:(80|443)/)) {
  urlHost = urlHost.replace(/:(80|443)/, '');
}
urlOrigin = url.protocol + '//' + urlHost;
```

Reply:
nice suggestion